src/simplestreams/simple_streams_manifest.cpp

```diff
@@ -113,21 +116,40 @@ std::unique_ptr<mp::SimpleStreamsManifest> mp::SimpleStreamsManifest::fromJson(c
             if (items.isEmpty())
                 continue;
 
-            const auto image = items["disk1.img"].toObject();
-            const auto image_location = image["path"].toString();
-            const auto sha256 = image["sha256"].toString();
-            const auto size = image["size"].toInt(-1);
-
-            // NOTE: These are not defined in the manifest itself
-            // so they are not guaranteed to be correct or exist in the server
-            const auto prefix = derive_unpacked_file_path_prefix_from(image_location);
-            const auto kernel_location = prefix + "-vmlinuz-generic";
-            const auto initrd_location = prefix + "-initrd-generic";
+            const auto& driver = utils::get_driver_str();
+
+            QJsonObject image;
+            QString sha256, image_location, kernel_location, initrd_location;
+            int size = -1;
+
+            if (driver == "lxd")
+            {
+                image = items["lxd.tar.xz"].toObject();
+                sha256 = image["combined_disk1-img_sha256"].toString();
+
+                if (sha256.isEmpty())
+                {
+                    continue;
+                }
```

Comment:
I think this?
```suggestion
                image = items["disk-kvm.img"].toObject();
                sha256 = image["sha256"].toString();
```

Reply:
Hmm, the `disk-kvm.img` is not available for 18.04 which is why I chose the `lxd.tar.xz`type.  And is LXD able to use that type at all?  I'm pretty sure LXD will not like it, but I can try.

Reply:
`lxd.tar.xz` is for containers, not VMs. I've asked what image types they're looking for.

Reply:
Hmm, I think both are valid for VM's.  Looking through the LXD code, it looks like they prefer the `disk-kvm.img` product, but if that is not available, it looks for `combined_disk1-img_sha256` under the `lxd.tar.xz` product.  I will do the same:)

Reply:
I stand corrected, they seem to be doing special stuff.